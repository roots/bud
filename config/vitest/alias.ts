import {path} from '@repo/constants'

export default {
  '@roots/bud': path(`sources/@roots/bud/src`),
  '@roots/bud-api': path(`sources/@roots/bud-api/src`),
  '@roots/bud-api/methods': path(`sources/@roots/bud-api/src/methods`),
  '@roots/bud-api/service': path(`sources/@roots/bud-api/src/service`),
  '@roots/bud-babel': path(`sources/@roots/bud-babel/src`),
  '@roots/bud-build': path(`sources/@roots/bud-build/src`),
  '@roots/bud-build/handlers': path(
    `sources/@roots/bud-build/src/handlers`,
  ),
  '@roots/bud-cache': path(`sources/@roots/bud-cache/src`),
  '@roots/bud-compiler': path(`sources/@roots/bud-compiler/src`),
  '@roots/bud-compress': path(`sources/@roots/bud-compress/src`),
  '@roots/bud-dashboard': path(`sources/@roots/bud-dashboard/src`),
  '@roots/bud-extensions': path(`sources/@roots/bud-extensions/src`),
  '@roots/bud-framework': path(`sources/@roots/bud-framework/src`),
  '@roots/bud-framework/extension': path(
    `sources/@roots/bud-framework/src/extension`,
  ),
  '@roots/bud-framework/extension/decorators': path(
    `sources/@roots/bud-framework/src/extension/decorators`,
  ),
  '@roots/bud-hooks': path(`sources/@roots/bud-hooks/src`),
  '@roots/bud-preset-react': path(`sources/@roots/bud-preset-react/src`),
  '@roots/bud-preset-recommend': path(
    `sources/@roots/bud-preset-recommend/src`,
  ),
  '@roots/bud-react': path(`sources/@roots/bud-react/src`),
  '@roots/bud-sass': path(`sources/@roots/bud-sass/src`),
  '@roots/bud-server': path(`sources/@roots/bud-server/src`),
  '@roots/bud-server/middleware': path(
    `sources/@roots/bud-server/src/middleware`,
  ),
  '@roots/bud-support': path(`sources/@roots/bud-support/src`),
  '@roots/bud-swc': path(`sources/@roots/bud-swc/src`),
  '@roots/bud-tailwindcss-theme-json': path(
    `sources/@roots/bud-tailwindcss-theme-json/src`,
  ),
  '@roots/bud-wordpress-externals': path(
    `sources/@roots/bud-wordpress-externals/src`,
  ),
  '@roots/bud/services': path(`sources/@roots/bud/src/services`),
  '@roots/container': path(`sources/@roots/container/src`),
  '@roots/entrypoints-webpack-plugin': path(
    `sources/@roots/entrypoints-webpack-plugin/src`,
  ),
  '@roots/sage': path(`sources/@roots/sage/src`),
  '@roots/wordpress-dependencies-webpack-plugin': path(
    `sources/@roots/wordpress-dependencies-webpack-plugin/src`,
  ),
  '@roots/wordpress-externals-webpack-plugin': path(
    `sources/@roots/wordpress-externals-webpack-plugin/src`,
  ),
  '@roots/wordpress-hmr': path(`sources/@roots/wordpress-hmr/src`),
  '@roots/wordpress-hmr/loader': path(
    `sources/@roots/wordpress-hmr/src/loader`,
  ),
}
