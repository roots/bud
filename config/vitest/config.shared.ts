import type {UserConfig} from 'vitest'

import {env} from 'node:process'

import GithubActionsReporter from 'vitest-github-actions-reporter'

export default {
  hookTimeout: 240000,
  reporters: env.GITHUB_ACTIONS
    ? [`basic`, new GithubActionsReporter()]
    : [`basic`],
  testTimeout: 240000,
  watch: false,
} satisfies UserConfig
