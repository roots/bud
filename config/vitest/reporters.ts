import {env} from 'node:process'

import GithubActionsReporter from 'vitest-github-actions-reporter'

const reporters: Array<any>  = [`basic`]
if (env.GITHUB_ACTIONS) reporters.push(new GithubActionsReporter())

export default reporters
