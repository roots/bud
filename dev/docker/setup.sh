#!/bin/bash

cd /bud

npx npm-cli-login -u test -p test -e test@test.com -r http://verdaccio:4873

yarn config set npmRegistryServer http://verdaccio:4873
yarn config set npmAuthIdent 'test:test'
yarn config set unsafeHttpWhitelist --json '["verdaccio"]'

echo Install
yarn

echo Build
yarn @bud build

echo Done!
