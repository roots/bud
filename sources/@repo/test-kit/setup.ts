import {type Options, default as Project} from './integration-test.js'

export default (options: Options) => new Project(options)

export {type Options, type Project}
