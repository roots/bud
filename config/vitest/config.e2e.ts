import {env} from 'node:process'

import GithubActionsReporter from 'vitest-github-actions-reporter'
import {defineConfig} from 'vitest/config'

import alias from './alias.js'

export default defineConfig({
  test: {
    alias,
    hookTimeout: 240000,
    include: [`tests/e2e/*.test.ts`],
    reporters: [
      `basic`,
      env.GITHUB_ACTIONS ? new GithubActionsReporter() : ``,
    ],
    testTimeout: 240000,
    threads: false,
  },
})
