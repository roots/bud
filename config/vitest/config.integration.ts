import {defineConfig} from 'vitest/config'

import shared from './config.shared'

export default defineConfig({
  test: {
    ...shared,
    include: [`tests/integration/**/*.test.ts`],
    pool: `forks`,
    poolOptions: {
      forks: {
        maxForks: 1,
        minForks: 1,
      },
    },
    setupFiles: [`./config/vitest/setup.integration.ts`],
    watch: false,
  },
})
