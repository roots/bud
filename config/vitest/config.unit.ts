import {env} from 'process'

import {defineConfig} from 'vitest/config'
import GithubActionsReporter from 'vitest-github-actions-reporter'

import alias from './alias.js'

export default defineConfig({
  test: {
    alias,
    coverage: {
      include: [`sources/@roots/*/src/**/*.{ts,tsx}`],
      provider: `v8`,
      reporter: [`text`, `html`],
    },
    exclude: [`sources/@repo/**/*`, `**/node_modules/**/*`],
    hookTimeout: 60000,
    include: [
      `sources/@roots/*/src/*.test.{ts,tsx}`,
      `sources/@roots/*/src/**/*.test.{ts,tsx}`,
      `sources/@roots/*/test/*.test.{ts,tsx}`,
      `sources/@roots/*/test/**/*.test.{ts,tsx}`,
      `sources/@roots/*/tests/*.test.{ts,tsx}`,
      `sources/@roots/*/tests/**/*.test.{ts,tsx}`,
      `tests/unit/**/*.test.ts`,
      `tests/reproductions/**/*.test.ts`,
      `tests/integration/*.test.ts`,
    ],
    includeSource: [`sources/@roots/*/src/**/*.{ts,tsx}`],
    reporters: [
      `default`,
      env.GITHUB_ACTIONS ? new GithubActionsReporter() : ``,
    ],
    testTimeout: 60000,
  },
})
