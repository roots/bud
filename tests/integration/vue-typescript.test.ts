import setup from '@repo/test-kit/setup'
import {describe, expect, it} from 'vitest'

describe(`examples/vue-typescript`, () => {
  it(`should compile assets as expected`, async () => {
    const test = setup({
      label: `@examples/vue-typescript`,
    })
    await test.install()
    await test.build()

    expect(test.manifest).toMatchSnapshot()
    expect(test.assets[`main.js`].length).toBeGreaterThan(10)
    expect(test.assets[`main.js`].includes(`from '`)).toBeFalsy()
    expect(test.assets[`main.css`].length).toBeGreaterThan(10)
    expect(test.assets[`main.css`].includes(`$vue-green`)).toBeFalsy()
  })
})
