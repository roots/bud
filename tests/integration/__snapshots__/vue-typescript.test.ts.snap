// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`examples/vue-typescript > should compile js and css as expected 1`] = `
{
  "main.css": "css/main.css",
  "main.js": "js/main.js",
}
`;
